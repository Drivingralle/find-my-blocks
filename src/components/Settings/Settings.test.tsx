import React from "react";
import renderer from "react-test-renderer";
import { cleanup, fireEvent, render } from "@testing-library/react";
import { Settings } from "./";

afterEach(cleanup);

it("render correctly", () => {
  const tree = renderer
    .create(
      <Settings onNavOrderChange={() => null} onCardOrderChange={() => null} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("returns nothing if `onNavOrderChange` is not set", () => {
  const onNavOrderChange = jest.fn();
  const { getByLabelText } = render(
    <Settings onCardOrderChange={() => null} />
  );

  const select = getByLabelText("Sort navigation:");
  fireEvent.change(select, {
    target: { value: "Test" },
  });

  expect(onNavOrderChange).toHaveBeenCalledTimes(0);
});

it("fires an event for `onNavOrderChange`", () => {
  const onNavOrderChange = jest.fn();
  const { getByLabelText } = render(
    <Settings
      onNavOrderChange={(val) => onNavOrderChange(val)}
      onCardOrderChange={() => null}
    />
  );

  const select = getByLabelText("Sort navigation:");
  fireEvent.change(select, {
    target: { value: "Test" },
  });

  expect(onNavOrderChange).toHaveBeenCalled();
});

it("returns nothing if `onCardOrderChange` is not set", () => {
  const onCardOrderChange = jest.fn();
  const { getByLabelText } = render(<Settings />);

  const select = getByLabelText("Sort cards:");
  fireEvent.change(select, {
    target: { value: "Test" },
  });

  expect(onCardOrderChange).toHaveBeenCalledTimes(0);
});

it("fires an event for `onCardOrderChange`", () => {
  const onCardOrderChange = jest.fn();
  const { getByLabelText } = render(
    <Settings onCardOrderChange={() => onCardOrderChange()} />
  );

  const select = getByLabelText("Sort cards:");
  fireEvent.change(select, {
    target: { value: "Test" },
  });

  expect(onCardOrderChange).toHaveBeenCalled();
});

it("fires an event for `onShowCoreBlocksClick`", () => {
  const onShowCoreBlocksClick = jest.fn();
  const { getByLabelText } = render(
    <Settings
      onShowCoreBlocksClick={() => onShowCoreBlocksClick()}
      onCardOrderChange={() => undefined}
      navOrder={""}
      cardOrder={""}
      showCoreBlocks={false}
      initialOpen={true}
      onNavOrderChange={() => undefined}
    />
  );

  const checkbox = getByLabelText("Show core blocks");
  fireEvent.click(checkbox);

  expect(onShowCoreBlocksClick).toHaveBeenCalled();
});

it("does nothing if `onShowCoreBlocksClick` not set", () => {
  const onShowCoreBlocksClick = jest.fn();
  const { getByLabelText } = render(
    // Ignoring for purpose of test
    // @ts-ignore
    <Settings
      onCardOrderChange={() => undefined}
      navOrder={""}
      cardOrder={""}
      showCoreBlocks={false}
      initialOpen={true}
      onNavOrderChange={() => undefined}
    />
  );

  const checkbox = getByLabelText("Show core blocks");
  fireEvent.click(checkbox);

  expect(onShowCoreBlocksClick).not.toHaveBeenCalled();
});
