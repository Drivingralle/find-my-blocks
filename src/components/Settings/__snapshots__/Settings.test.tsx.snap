// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div
  className="components-panel"
>
  <div
    className="components-panel__body is-opened"
  >
    <h2
      className="components-panel__body-title"
    >
      <button
        aria-expanded={true}
        className="components-button components-panel__body-toggle"
        onClick={[Function]}
        type="button"
      >
        <span
          aria-hidden="true"
        >
          <svg
            aria-hidden={true}
            className="components-panel__arrow"
            focusable={false}
            height={24}
            role="img"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 8l-6 5.4 1 1.2 5-4.6 5 4.6 1-1.2z"
            />
          </svg>
        </span>
        Settings
        <svg
          aria-hidden={true}
          className="components-panel__icon"
          focusable={false}
          height={20}
          role="img"
          viewBox="0 0 24 24"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.4 12c0-.5 0-.9-.1-1.3l1.4-1.3-1.5-2.8-1.8.6c-.6-.6-1.4-1-2.2-1.3L13.8 4h-3l-.4 1.9c-.8.3-1.5.8-2.1 1.3l-1.8-.6L5 9.4l1.4 1.3c-.1.4-.1.9-.1 1.3s0 .9.1 1.3L5 14.6l1.5 2.8 1.8-.6c.6.6 1.4 1 2.2 1.3l.4 1.9h3l.4-1.9c.8-.3 1.5-.7 2.2-1.3l1.8.6 1.5-2.8-1.4-1.3c-.1-.4 0-.8 0-1.3zm-6.1 3.2c-1.7 0-3-1.4-3-3.2 0-1.8 1.4-3.2 3-3.2 1.7 0 3 1.4 3 3.2.1 1.8-1.3 3.2-3 3.2z"
          />
        </svg>
      </button>
    </h2>
    <div
      className="components-panel__row"
    >
      <div
        className="components-base-control select"
      >
        <div
          className="components-base-control__field"
        >
          <label
            className="components-base-control__label"
            htmlFor="inspector-select-control-0"
          >
            Sort navigation:
          </label>
          <select
            className="components-select-control__input"
            id="inspector-select-control-0"
            multiple={false}
            onChange={[Function]}
          >
            <option
              value="az"
            >
              Alphabetically (A-Z)
            </option>
            <option
              value="za"
            >
              Alphabetically (Z-A)
            </option>
            <option
              value="high-low"
            >
              Most Popular
            </option>
            <option
              value="low-high"
            >
              Least Popular
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      className="components-panel__row"
    >
      <div
        className="components-base-control select"
      >
        <div
          className="components-base-control__field"
        >
          <label
            className="components-base-control__label"
            htmlFor="inspector-select-control-1"
          >
            Sort cards:
          </label>
          <select
            className="components-select-control__input"
            id="inspector-select-control-1"
            multiple={false}
            onChange={[Function]}
          >
            <option
              value="az"
            >
              Alphabetically (A-Z)
            </option>
            <option
              value="za"
            >
              Alphabetically (Z-A)
            </option>
            <option
              value="popular"
            >
              Most Popular
            </option>
            <option
              value="reusable"
            >
              Reusable First
            </option>
          </select>
        </div>
      </div>
    </div>
    <h4>
      Show/hide block types
    </h4>
    <div
      className="components-base-control"
    >
      <div
        className="components-base-control__field"
      >
        <span
          className="components-checkbox-control__input-container"
        >
          <input
            className="components-checkbox-control__input"
            id="inspector-checkbox-control-0"
            onChange={[Function]}
            type="checkbox"
            value="1"
          />
        </span>
        <label
          className="components-checkbox-control__label"
          htmlFor="inspector-checkbox-control-0"
        >
          Show core blocks
        </label>
      </div>
    </div>
  </div>
</div>
`;
